crates/types/protobuf/restate/common.proto

```diff
@@ -33,15 +33,14 @@ message Version { uint32 value = 1; }
 // The handle name or type tag of the message. For every target there must be
 // exactly one message handler implementation.
 enum TargetName {
+  reserved 7, 8;
   TargetName_UNKNOWN = 0;
   METADATA_MANAGER = 1;
   INGRESS = 2;
   LOCAL_METADATA_STORE = 3;
   LOCAL_METADATA_STORE_CLIENT = 4;
   ATTACH_REQUEST = 5;
   ATTACH_RESPONSE = 6;
-  GET_PROCESSORS_STATE_REQUEST = 7;
-  PROCESSORS_STATE_RESPONSE = 8;
```

Comment:
It's a very good question.

I tried to follow the protobuf guide lines for deprecating enum variants and added the `reserved 7,8` tagging. My worries if we leave them around that they can be reused by accident or intentionally which might cause unforeseen issues (maybe)

My thought was:
- A newer node will  never send a `GET_PROCESSORS_STATE_REQUET` message hence there will be no issue
- An older node can still send this, but newer nodes that will receive this request will log a warning and drop the message. 
 - Supposedly this will happen for a short time until all nodes are updated to latest version

Reply:
@pcholakov will `restatectl` send these messages or would it talk to the `Admin` component or directly to the node via grpc?

I think in this version it is ok to remove it since we won't have the situation that we have to deal with a cluster of mixed version nodes. In the future, this might be different. Then, we can only remove messages that are no longer used by the current and previous version. Otherwise, we risk that the cluster stops working.